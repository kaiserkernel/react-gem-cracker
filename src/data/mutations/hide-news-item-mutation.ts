export const hideNewsItemMutation = `
  mutation HideNewsItem($id: Int!) {
    hideNewsItem(id: $id) {
      id
      hidden
    }
  }
`;
