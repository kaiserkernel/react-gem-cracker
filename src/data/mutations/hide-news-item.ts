import { gql } from 'apollo-server-express';

export const hideNewsItem = gql`
  mutation HideNewsItem($id: Int!) {
    hideNewsItem(id: $id) {
      id
      hidden
    }
  }
`;
