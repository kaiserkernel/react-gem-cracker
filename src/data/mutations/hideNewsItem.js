import gql from 'graphql-tag';

export default gql`
  mutation HideNewsItem($id: Int!) {
    hideNewsItem(id: $id) {
      id
      hidden
    }
  }
`;
