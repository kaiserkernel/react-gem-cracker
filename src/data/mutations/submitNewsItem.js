import gql from 'graphql-tag';
import NewsFeed from '../../components/presentational/NewsFeed';

export default gql`
  mutation SubmitNewsItem($title: String!, $url: String) {
    submitNewsItem(title: $title, url: $url) {
      id
      ...NewsFeed
    }
  }
  ${NewsFeed.fragments.newsItem}
`;
