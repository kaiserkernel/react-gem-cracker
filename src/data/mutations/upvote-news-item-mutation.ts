export const upvoteNewsItemMutation = `
  mutation UpvoteNewsItem($id: Int!) {
    upvoteNewsItem(id: $id) {
      id
      upvoteCount
      upvoted
    }
  }
`;
