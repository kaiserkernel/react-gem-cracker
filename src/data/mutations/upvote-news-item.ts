export const upvoteNewsItem = `
  mutation UpvoteNewsItem($id: Int!) {
    upvoteNewsItem(id: $id) {
      id
      upvoteCount
      upvoted
    }
  }
`;
