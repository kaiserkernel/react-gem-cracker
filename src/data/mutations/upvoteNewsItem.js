import { gql } from 'react-apollo';

export default gql`
  mutation UpvoteNewsItem($id: Int!) {
    upvoteNewsItem(id: $id) {
      id
      upvoteCount
      upvoted
    }
  }
`;
