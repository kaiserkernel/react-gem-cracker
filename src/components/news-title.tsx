import gql from 'graphql-tag';
import Router from 'next/router';
import * as React from 'react';
import { graphql } from 'react-apollo';
import { parse } from 'url';

import { upvoteNewsItem } from '../data/mutations/upvote-news-item';

export interface INewsTitleProps {
  id: number;
  isRankVisible?: boolean;
  isUpvoteVisible?: boolean;
  rank?: number;
  title: string;
  url?: string;
  upvoted: boolean;
  upvoteNewsItem: (id: number) => void;
}

export class NewsTitleView extends React.Component<INewsTitleProps> {
  static fragments = {
    newsItem: gql`
      fragment NewsTitle on NewsItem {
        id
        title
        url
        upvoted
      }
    `,
  };

  static defaultProps = {
    isRankVisible: true,
    isUpvoteVisible: true,
    rank: undefined,
    url: undefined,
  };

  render(): JSX.Element {
    return (
      <tr className="athing">
        <td style={{ textAlign: 'right', verticalAlign: 'top' }} className="title">
          <span className="rank">{this.props.isRankVisible && `${this.props.rank}.`}</span>
        </td>
        <td style={{ verticalAlign: 'top' }} className="votelinks">
          <div style={{ textAlign: 'center' }}>
            {this.props.isUpvoteVisible && (
              <a
                className={this.props.upvoted ? 'nosee' : ' '}
                onClick={() => this.props.upvoteNewsItem(this.props.id)}
                href="javascript:void(0)"
              >
                <div className="votearrow" title="upvote" />
              </a>
            )}
          </div>
        </td>
        <td className="title">
          <a className="storylink" href={this.props.url ? this.props.url : `item?id=${this.props.id}`}>
            {this.props.title}
          </a>
          {this.props.url && (
            <span className="sitebit comhead">
              {' '}
              (
              <a href={`from?site=${parse(this.props.url).hostname}`}>
                <span className="sitestr">{parse(this.props.url).hostname}</span>
              </a>
              )
            </span>
          )}
        </td>
      </tr>
    );
  }
}

export const NewsTitle = graphql(upvoteNewsItem, {
  props: ({ ownProps, mutate }) => ({
    upvoteNewsItem: id =>
      mutate({
        variables: { id },
      })
        // .then(() => Router.push(`/login?id=${id}&password=${password}`))
        .catch(() => Router.push('/login', `/vote?id=${id}&how=up&goto=news`)),
  }),
})(NewsTitleView);
