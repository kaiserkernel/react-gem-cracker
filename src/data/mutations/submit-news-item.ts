import gql from 'graphql-tag';
import { NewsFeedView } from '../../components/news-feed';

export const submitNewsItem = gql`
  mutation SubmitNewsItem($title: String!, $url: String) {
    submitNewsItem(title: $title, url: $url) {
      id
      ...NewsFeed
    }
  }
  ${NewsFeedView.fragments.newsItem}
`;
