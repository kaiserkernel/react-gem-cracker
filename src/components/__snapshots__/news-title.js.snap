// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsTitle component renders news item properties passed in as props 1`] = `
<tr
  className="athing"
  id={1224}
>
  <td
    className="title"
    style={
      Object {
        "textAlign": "right",
        "verticalAlign": "top",
      }
    }
  >
    <span
      className="rank"
    >
      1.
    </span>
  </td>
  <td
    className="votelinks"
    style={
      Object {
        "verticalAlign": "top",
      }
    }
  >
    <center>
      <a
        className=" "
        href="javascript:void(0)"
        onClick={[Function]}
      >
        <div
          className="votearrow"
          title="upvote"
        />
      </a>
    </center>
  </td>
  <td
    className="title"
  >
    <a
      className="storylink"
      href="https://www.nytimes.com/2017/08/04/education/edlife/learning-how-to-learn-barbara-oakley.html?hpw&rref=education&action=click&pgtype=Homepage&module=well-region&region=bottom-well&WT.nav=bottom-well"
    >
      “Learning How to Learn,” the most popular course on Coursera
    </a>
    <span
      className="sitebit comhead"
    >
       (
      <a
        href="from?site=www.nytimes.com"
      >
        <span
          className="sitestr"
        >
          www.nytimes.com
        </span>
      </a>
      )
    </span>
  </td>
</tr>
`;
